### PR TITLE
V2 us 05 merchant dashboard invoices sorted by least recent

### LINKED ISSUE #5: US-04-merchant-dashboard-items-ready-to-ship
4. Merchant Dashboard Items Ready to Ship

As a merchant
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see a section for "Items Ready to Ship"
In that section I see a list of the names of all of my items that
have been ordered and have not yet been shipped,
And next to each Item I see the id of the invoice that ordered my item
And each invoice id is a link to my merchant's invoice show page

### LINKED ISSUE #64: V2 us 05 merchant dashboard invoices sorted by least recent
update date_invoice_created

### PR DESCRIPTION
**Pull request by yainporter**
## Type of change
- [ ] New feature
- [X] Bug Fix
- [ ] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s): 
US-5
*Brief summary of the change(s) included in this pull request*
- Changed the name of #date_invoice_created to be more descriptive => #date_an_invoice_was_created
- Made sure that the method #date_an_invoice_was_created takes in a parameter so we can link an Item with an Invoice

## Implements/Fixes Issue(s):
closes #64

## Neccesary checkmarks:
- [X] All Tests are Passing
- [X] The code will run locally

## Any Outstanding Issues/Problems?
- [ ] My code is wonderful, no issues here!
- [X] Yeah, see additional words below
*Description of problem(s)*
```
 def date_an_invoice_was_created(invoice_id)
    invoices.exists? ? invoices.find(invoice_id).format_date_created : "No Invoice for this Item"
  end
```
Could probably be refactored in the future, but I think it is fine as it is:
- finds an Item's matching invoice
- formats the date if invoices exist for an item, otherwise returns "No Invoice for this Item"

## Any Fun Facts or Gifs You Want to Add to This PR?

Find this delicious ice cream at the grocery store, and enjoy

### CODE REVIEW

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -40,23 +40,20 @@
...
-        expect(@item_1.date_invoice_created).to eq("Thursday, September 30, 2021")
-        expect(@item_2.date_invoice_created).to eq("Saturday, October 12, 2019")
-
-        item = create(:item)
-
-        expect(item.date_invoice_created).to eq("No Invoice for this Item")
+    describe "date_an_invoice_was_created" do
+      it "returns the created_at date for a particular invoice of an item" do
+        expect(@item_1.date_an_invoice_was_created(@invoice_1.id)).to eq("Thursday, September 30, 2021")
+        expect(@item_2.date_an_invoice_was_created(@invoice_2.id)).to eq("Saturday, October 12, 2019")
```
Comment (kohljd) [inline spec/models/item_spec.rb:56]: Why was the test for "No Invoice for the Item"? It doesn't look like it was taken out of the method

File: spec/models/merchant_spec.rb [hunk 2]
```diff
@@ -14,63 +14,50 @@
...
-    @invoice_2 = create(:invoice, customer: @customer_2)
-    @invoice_3 = create(:invoice, customer: @customer_3)
-    @invoice_4 = create(:invoice, customer: @customer_4)
-    @invoice_5 = create(:invoice, customer: @customer_5)
-    @invoice_6 = create(:invoice, customer: @customer_6)
+    create_list(:transaction, 20, invoice_id: @invoice_5.id)
+    create_list(:transaction, 15, invoice_id: @invoice_2.id)
+    create_list(:transaction, 10, invoice_id: @invoice_1.id)
+    create_list(:transaction, 7, invoice_id: @invoice_3.id)
+    create_list(:transaction, 5, invoice_id: @invoice_4.id)
```
Comment (kohljd) [inline spec/models/merchant_spec.rb:34]: Like the create_lists!

**Review by kohljd — CHANGES_REQUESTED** [review]
It looks like there were more changes than just a bug fix

File: db/seeds.rb
[the remark this answers is not in the document]
Comment (yainporter) [reply db/seeds.rb:1]: We will eventually replace this with our rake task, just wanted it here so I didn't have to keep making it, good catch!

File: spec/models/item_spec.rb (hunk 1 above)
Comment (yainporter) [reply spec/models/item_spec.rb:56]: I missed this, good catch! Also... when comments are in just the PR page, it doesn't display the whole file, I was so confused about what you were saying

File: spec/models/merchant_spec.rb (hunk 2 above)
Comment (yainporter) [reply spec/models/merchant_spec.rb:34]: :P

File: spec/models/item_spec.rb (hunk 1 above)
Comment (kohljd) [reply spec/models/item_spec.rb:56]: Thanks for adding that test back in!

**Review by kohljd — APPROVED** [review]

**Pull request merged**